src/fondant/core/component_spec.py

```diff
@@ -109,6 +109,48 @@ def to_file(self, path) -> None:
         with open(path, "w", encoding="utf-8") as file_:
             yaml.dump(self._specification, file_)
 
+    @classmethod
+    def from_dict(cls, component_spec_dict: t.Dict[str, t.Any]) -> "ComponentSpec":
+        """Load the component spec from a dictionary."""
+        return cls(component_spec_dict)
+
+    @classmethod
+    def from_args(
```

Comment:
I'm still in favor of making this the `__init__` method, and storing the different parts of the spec as different attributes instead of converting to a spec.